runtime/lua/vim/filetype/detect.lua

```diff
@@ -43,10 +43,27 @@ function M.btm(bufnr)
   end
 end
 
-function M.cfg(path, bufnr)
+-- Returns true if file content looks like RAPID
+local function is_rapid(bufnr, extension)
+  if extension == "cfg" then
+    local line = getlines(bufnr, 1):lower()
+    return findany(line, { "eio:cfg", "mmc:cfg", "moc:cfg", "proc:cfg", "sio:cfg", "sys:cfg" })
+  end
+  local first = "^%s*module%s+%S+%s*"
+  -- Called from mod, prg or sys functions
+  for _, line in ipairs(getlines(bufnr, 1, -1)) do
+    if not line:find("^%s*$") then
+      return findany(line:lower(), { "^%s*%%%%%%", first .. "(", first .. "$" })
```

Comment:
@gpanders I hope it's fine that I implemented this test function in this PR ;) Do you think I understood `\%{3}` correctly in the Vim regex: (`\v^\s*%(\%{3}|module\s+\k+\s*%(\(|$))`)? I think these are 3 percentage signs but not 100% sure.

Reply:
>Do you think I understood \%{3} correctly in the Vim regex: (\v^\s*%(\%{3}|module\s+\k+\s*%(\(|$)))? I think these are 3 percentage signs but not 100% sure.

Yes that is correct.